### PR TITLE
PROGRAMMES-6228 Adding credits to episode page

### PR DESCRIPTION
**Pull request by votemike**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6228

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by votemike** [comment]
Tests are for babies

File: templates/find_by_pid/episode.html.twig [hunk 1]
```diff
@@ -6,53 +6,97 @@
...
+                        <div class="component__header br-box-page">
+                            <h2>{{ tr('credits', programme.getContributionsCount()) }}</h2>
+                        </div>
+                        <div class="component__body br-box-page">
+                            <div>
+                                <table class="table table--slatted-vertical no-margin-vertical">
+                                    <thead class="visually-hidden">
+                                    <tr>
+                                        {% if showRole %}
+                                            <th>Role</th>
```
Comment (votemike) [inline templates/find_by_pid/episode.html.twig]: No translations in the translations file...

File: templates/find_by_pid/episode.html.twig [hunk 2]
```diff
@@ -6,53 +6,97 @@
...
+                                        {% endif %}
+                                        <th>Contributor</th>
```
Comment (votemike) [inline templates/find_by_pid/episode.html.twig]: No translations in the translations file...

File: templates/find_by_pid/episode.html.twig [hunk 3]
```diff
@@ -6,53 +6,95 @@
...
+                            </div>
+                            <div class="component__body br-box-page">
+                                <div>
+                                    <table class="table table--slatted-vertical no-margin-vertical">
+                                        <thead class="visually-hidden">
+                                        <tr>
+                                            {% if showRole %}
+                                                <th>Role</th>
+                                            {% endif %}
+                                            <th>Contributor</th>
```
Comment (votemike) [inline templates/find_by_pid/episode.html.twig]: No translations in the translations file...

**Comment by BjornTwachtmann** [comment]
In V2 the credits partial is used in a couple of places, including the clip and episode pages. I think I'd move it into its own presenter, also avoids having to have the view logic for shouldShowRole in the controller.

File: src/Ds2013/Presenters/Utilities/Credits/credits.html.twig [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<table class="table table--slatted-vertical no-margin-vertical">
+    <thead class="visually-hidden">
+    <tr>
+        {% if credits.shouldShowRoleColumn() %}
+            <th>Role</th>
```
Comment (votemike) [inline src/Ds2013/Presenters/Utilities/Credits/credits.html.twig]: There don't seem to be any translations for this.... Is that OK?

File: src/Ds2013/Presenters/Utilities/Credits/credits.html.twig [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        {% endif %}
+        <th>Contributor</th>
```
Comment (votemike) [inline src/Ds2013/Presenters/Utilities/Credits/credits.html.twig]: There don't seem to be any translations for this.... Is that OK?

File: templates/find_by_pid/episode.html.twig [hunk 6]
```diff
@@ -6,53 +6,65 @@
...
 
-                {# TODO Credits module #}
+                    {% if programme.getContributionsCount() %}
+                        <div class="component component--box component--box-flushbody-vertical component--box--primary"
+                             id="credits">
+                            <div class="component__header br-box-page">
+                                <h2>{{ tr('credits', programme.getContributionsCount()) }}</h2>
+                            </div>
+                            <div class="component__body br-box-page">
+                                <div>
```
Comment (votemike) [inline templates/find_by_pid/episode.html.twig]: The <div> seems superfluous. Should I get rid of it? Or does someone know why it necessary?

File: src/Ds2013/Presenters/Utilities/Credits/credits.html.twig (hunk 4 above)
Comment (BjornTwachtmann) [reply src/Ds2013/Presenters/Utilities/Credits/credits.html.twig]: We don't currently have any translations. It's probably worth raising a ticket and sticking it on the backlog, but this is fine for now.

Comment (votemike) [reply src/Ds2013/Presenters/Utilities/Credits/credits.html.twig]: https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6243

**Pull request merged**